File: Leetcode/XOR operation/Xoroperation.java
import java.io.*;
class Xoroperation {
    public int xorOperation(int n, int start) {
        // Function to find the XOR of 
    // all elements in the array 
        int[] ar = new int[n];
     // Iterating through every element in 
        // the array
        for(int i = 0; i < n; i++) {
            ar[i] = start + 2*i;
        }

        int result = start;
        for(int i = 0; i < n-1; i++) {
            result = result ^ ar[i+1];
        }
    //Return the XOR 
        return result;  
    }
}

File: Leetcode/XOR operation/XOR_Operation.py
from functools import reduce
class Solution:
    def xorOperation(self,n: int, start: int) -> int:
        nums=[]  #list to store numbers based on the given condition
        for i in range(n):  #iterates from 0 to n-1
            m=start+(2*i)
            nums.append(m) #adding every k at the end of num 
            
        res = reduce(lambda x, y: x ^ y, nums) #applying Xor operation and calculates like((((0^2)^4)^6)^8).
        return(res)
 
